app/services/theme.js

```diff
@@ -1,42 +1,44 @@
 import Ember from 'ember';
 import config from 'ember-get-config';
+import { pluralize } from 'ember-inflector';
+
 
-/**
- * @module ember-osf-reviews
- * @submodule services
- */
-
-/**
- * Allows you to inject that provider's theme into parts of your application
- *
- * @class theme
- * @extends Ember.Service
- */
 export default Ember.Service.extend({
+    i18n: Ember.inject.service(),
     store: Ember.inject.service(),
-    session: Ember.inject.service(),
-
-    // If we're using a provider domain
-    isDomain: window.isProviderDomain,
 
-    // The id of the current provider
-    id: config.Reviews.defaultProvider,
+    provider: null,
 
-    provider: Ember.computed('id', function() {
-        return this.get('store').findRecord('preprint-provider', this.get('id'));
-    }),
+    id: Ember.computed.alias('provider.id'),
+    isLoaded: Ember.computed.empty('provider'),
+    isProvider: Ember.computed.not('isNotProvider'),
+    isNotProvider: Ember.computed.equal('provider.id', 'OSF'),
 
-    isProvider: Ember.computed('id', function() {
-        return this.get('id') !== 'osf';
-    }),
-
-    // The url to redirect users to sign up to
     signupUrl: Ember.computed('id', function() {
         const query = Ember.$.param({
             campaign: `${this.get('id')}-reviews`,
             next: window.location.href
         });
 
         return `${config.OSF.url}register?${query}`;
-    })
+    }),
+
+    onProviderLoad: Ember.observer('provider', function() {
```

Comment:
These should be in `ember-osf`'s `theme`, so `ember-preprints` can use it too

Reply:
From Ember OSF:
> Stub so discover page can check for a theme.  Override on your consuming application.

Reply:
Right, but since you tore out the `preprintWords` translation stuff from ember-osf, something needs to replace it. It seems reasonable to make `theme` slightly more than a stub. Unless we're planning to duplicate this logic in our ember-preprints branch? Or move the old `preprintWords` logic there?